src/agent/application.cpp

```diff
@@ -389,7 +395,30 @@ void Application::InitNcpMode(void)
         {
             mBorderAgentUdpProxy.SendToPeer(aUdpPayload, aLength, aPeerAddr, aPeerPort);
         }
+#if OTBR_ENABLE_EPSKC
+        else if (aLocalPort == mEphemeralKeyUdpProxy.GetThreadPort())
+        {
+            mEphemeralKeyUdpProxy.SendToPeer(aUdpPayload, aLength, aPeerAddr, aPeerPort);
+        }
+#endif // OTBR_ENABLE_EPSKC
+    });
+#if OTBR_ENABLE_EPSKC
+    mHost.AddEphemeralKeyStateChangedCallback([this](otBorderAgentEphemeralKeyState aState, uint16_t aPort) {
+        if (aState == OT_BORDER_AGENT_STATE_STARTED)
+        {
+            otbrLogInfo("Border Agent Ephemeral Key State Changed: Active on port %d", aPort);
+            mEphemeralKeyUdpProxy.Start(aPort);
+        }
+        else if (aState == OT_BORDER_AGENT_STATE_STOPPED || aState == OT_BORDER_AGENT_STATE_DISABLED)
+        {
+            otbrLogInfo("Border Agent Ephemeral Key State Changed: Inactive");
+            mEphemeralKeyUdpProxy.Stop();
+#if OTBR_ENABLE_BORDER_AGENT_MESHCOP_SERVICE
+            mBorderAgent.HandleEpskcStateChanged(aState, aPort);
+#endif // OTBR_ENABLE_BORDER_AGENT_MESHCOP_SERVICE
```

Comment:
This should be invoked in both cases? When state is `STARTED`, this should also be called to publish the meshcop-e service.